Lib/test/test_weakref.py

```diff
@@ -1609,6 +1609,43 @@ def test_weak_valued_dict_update(self):
             self.assertEqual(list(d.keys()), [kw])
             self.assertEqual(d[kw], o)
 
+    def test_weak_valued_union_operators(self):
+        a = C()
+        b = C()
+        c = C()
+        wvd1 = weakref.WeakValueDictionary({1 : a})
+        wvd2 = weakref.WeakValueDictionary({1 : b, 2 : a})
+        wvd3 = wvd1.copy()
+        d1 = {1 : c, 3 : b}
+        pairs = [(5, c), (5, b)]
```

Comment:
This is an interesting test case, but I just want to verify that you intended to use the same key twice here.

You didn't mean something like this, which results in a mapping of length 2?

```suggestion
        pairs = [(5, c), (6, b)]
```